modules/generic/src/main/scala/io/chrisdavenport/cormorant/generic/auto.scala

```diff
@@ -8,7 +8,7 @@ import shapeless.labelled._
  * Fully Automatic Derivation of Any Product Type
   **/
 object auto {
-  implicit val hnilWrite = semiauto.hnilWrite
+  implicit def hnilWrite[H](implicit P: Put[H]): Write[H :: HNil] = semiauto.hnilWrite[H]
```

Comment:
Is it possible / sensible to add an ambiguous implicit for `HNil` by itself, with an implicitNotFound message pointing out the non-empty portion?

Reply:
Due to the model of NonEmptyList, `HNil` by itself no longer makes much sense, instead we need to match on the last element that would be written and assert that this works and finishes off the HList.

I'm very open to better implicit not found instances.

Reply:
Right, the suggestion is to provide an *ambiguous* implicit for HNil to prompt scala to give the message. I suppose in retrospect it's not useful, as an empty type would simply fail to find one due to not having one for HNil